pkg/scheduler/scheduler.go

```diff
@@ -36,22 +36,25 @@ import (
 	kueue "sigs.k8s.io/kueue/api/v1alpha1"
 	"sigs.k8s.io/kueue/pkg/cache"
 	"sigs.k8s.io/kueue/pkg/queue"
+	"sigs.k8s.io/kueue/pkg/util/function"
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
 type Scheduler struct {
-	queues   *queue.Manager
-	cache    *cache.Cache
-	client   client.Client
-	recorder record.EventRecorder
+	queues     *queue.Manager
+	cache      *cache.Cache
+	client     client.Client
+	recorder   record.EventRecorder
+	funcLogger function.Logger
```

Comment:
`admissionRoutineWrapper`

but I don't see anything wrong just having a pointer to a `WaitGroup` here instead. Then, it would be called `admissionRoutineWG`.

Reply:
Using a pointer to `WaitGroup` is absolutely right... and I just want to make it more extensible by using a `wrapper` cause only in unit tests do we need this `WaitGroup`.